### DISCUSSION TITLE
Improve the performance of DeleteMultipleObjects

### DISCUSSION THREAD

**Discussion by chenshi5012** [minio/minio#18091] [Q&A] [answered]
we use minio for flink task checkpoint.
the  request to DeleteMultipleObjects too frequently， so we find that  DeleteMultipleObjects  request  keep long in stock 。

minio_s3_requests_inflight_total
![image](https://github.com/minio/minio/assets/10794822/b9668389-a817-4f98-aee1-dfc4fc3761fa)

minio_s3_requests_waiting_total
![image](https://github.com/minio/minio/assets/10794822/6e63f4f3-70bd-4924-8361-4c058df27fa9)

response time DeleteMultipleObjects keep at 5miniues， but the upload/download Not affected ，io not busy， but  the minio cpu  keep at 100%
![image](https://github.com/minio/minio/assets/10794822/905f100c-c895-4ed1-a104-bc8bc1c28eff)
![image](https://github.com/minio/minio/assets/10794822/1886050f-50f0-4eb3-a54f-445536c0fe6d)

is there any lock for DeleteMultipleObjects ？  is there any configure to improve the performance ?

**Comment by Jazz9** [comment]
We have this same issue and I haven't found a fix.

**Comment by klauspost** [comment, marked as answer]
It is a bulk request, potentially equivalent to thousands of individual requests. If your hardware cannot keep up, it will be slow.

**Reply by Jazz9** [reply]
Yes - this is true. Our system is multi-tenanted and we use the request limiter in minio to slow clients / make it fair. One multi-delete with 1000 deletes should be 1000 requests against the limiter and not 1. Currently, any client can issue enough multi-deletes to flood any hdd minio.
